src/yaml/pull_parser.cr

```diff
@@ -22,58 +45,146 @@ class YAML::PullParser
     end
 
     read_next
-    raise "Expected STREAM_START" unless kind == LibYAML::EventType::STREAM_START
+    raise "Expected STREAM_START" unless kind.stream_start?
   end
 
+  # Creates a parser, yields it to the block, and closes
+  # the parser at the end of it.
   def self.new(content)
     parser = new(content)
     yield parser ensure parser.close
   end
 
-  def kind
+  # Associates an object with an anchor, if this pull parser
+  # has an anchor at this point.
+  def record_anchor(object) : Nil
+    return unless object.is_a?(Reference)
+
+    record_anchor(object.object_id, object.crystal_type_id)
+  end
+
+  private def record_anchor(object_id, crystal_type_id)
+    anchor = self.anchor
+    return unless anchor
```

Comment:
Why not simplify it to:

```cr
return unless anchor = self.anchor
```

Reply:
I find it harder to understand like that.